packages/lesswrong/components/posts/PostsPage/PostsPage.tsx

```diff
@@ -81,6 +84,74 @@ export const getPostDescription = (post: {contents?: {plaintextDescription: stri
   return null;
 };
 
+/**
+ * Build the structured data for a post for SEO purposes.
+ * See https://developers.google.com/search/docs/appearance/structured-data/intro-structured-data for more info.
+ */
+const getStructuredData = ({
+  post,
+  description,
+}: {
+  post: PostsWithNavigation | PostsWithNavigationAndRevision;
+  description: string | null;
+}) => {
+  const hasUser = !!post.user;
+  const hasCoauthors = !!post.coauthors && post.coauthors.length > 0;
+
+  return {
+    "@context": "http://schema.org",
+    "@type": "DiscussionForumPosting",
+    mainEntityOfPage: {
+      "@type": "WebPage",
+      "@id": postGetPageUrl(post, true),
+    },
+    headline: post.title,
+    ...(description && { description: description }),
+    datePublished: new Date(post.postedAt).toISOString(),
+    about: post.tags.filter(tag => !!tag.description?.htmlHighlight).map(tag => ({
+      "@type": "Thing",
+      name: tag.name,
+      url: tagGetUrl(tag, undefined, true),
+      description: tag.description?.htmlHighlight,
```

Comment:
ChatGPT informs me that google should be smart about extracting useful information from this html (the full plaintextDescription isn't available, and I don't want to add it for performance reasons). I'll check the search results in a couple of weeks and see if this is the case, and remove it if it e.g. includes raw html in the results